src/utils/lm/install.js

```diff
@@ -0,0 +1,262 @@
+const os = require('os')
+const path = require('path')
+const process = require('process')
+
+const chalk = require('chalk')
+const execa = require('execa')
+const hasbin = require('hasbin')
+const Listr = require('listr')
+const tempDir = require('temp-dir')
+
+const { shouldFetchLatestVersion, fetchLatestVersion } = require('../../lib/exec-fetcher')
+const { fileExistsAsync, writeFileAsync, mkdtempAsync, readFileAsync, appendFileAsync } = require('../../lib/fs')
+const { getLegacyPathInHome } = require('../../lib/settings')
+
+const PACKAGE_NAME = 'netlify-credential-helper'
+const EXEC_NAME = 'git-credential-netlify'
+
+const { checkGitVersionStep, checkGitLFSVersionStep, checkLFSFiltersStep } = require('./steps')
+
+const getSetupStep = (skipInstall) => {
+  const platform = os.platform()
+  switch (platform) {
+    case 'linux':
+      return setupUnix('Linux', skipInstall)
+    case 'darwin':
+      return setupUnix('Mac OS X', skipInstall)
+    case 'win32':
+      return setupWindows(skipInstall)
+    default:
+      throw new Error(`Platform not supported: ${platform}.
+See manual setup instructions in https://github.com/netlify/netlify-credential-helper#install`)
+  }
+}
+
+const setupGitConfig = async function () {
+  return await configureGitConfig(getHelperPath())
+}
+
+const setupGitConfigStep = {
+  title: `Configuring Git to use Netlify's Git Credential Helper`,
+  task: setupGitConfig,
+}
+
+const installPlatform = async function (force) {
+  const skipInstall = !force && (await installedWithPackageManager())
+  const steps = [
+    checkGitVersionStep,
+    checkGitLFSVersionStep,
+    checkLFSFiltersStep(async (ctx, task, installed) => {
+      if (!installed) {
+        await execa('git', ['lfs', 'install'])
+        task.title += chalk.dim(' [installed]')
+      }
+    }),
+    getSetupStep(skipInstall),
+    setupGitConfigStep,
+  ]
+
+  const tasks = new Listr(steps)
+  await tasks.run()
+
+  return !skipInstall
+}
+
+const skipHelperInstall = function (skip) {
+  if (skip) {
+    return `Netlify's Git Credential Helper already installed with a package manager`
+  }
+}
+
+const installedWithPackageManager = async function () {
+  const installed = hasbin.sync('git-credential-netlify')
+  return installed && !(await fileExistsAsync(getBinPath()))
+}
+
+const setupWindows = function (skipInstall) {
+  return {
+    title: `Installing Netlify's Git Credential Helper for Windows`,
+    skip: () => skipHelperInstall(skipInstall),
+    task: installWithPowershell,
+  }
+}
+
+const installUnixHelper = async function () {
```

Comment:
Most of the code in that file is due to adding some binary to the `PATH`. This requires detecting shells, having shell-specific code, etc.

Is there a potential alternative where we would not need to modify the `PATH`? For example:
  - Calling the binary ourselves (so we can use an absolute/relative file path instead of the `PATH`)
  - Using a Node module + `npx`
  - Using a Node module and globally installing it

Note that those solutions might not make sense here.

Reply:
I believe we install that binary to be used by `git-lfs`. I'll verify that and also check if there's a better alternative

Reply:
Per `git` [documentation](https://git-scm.com/docs/gitcredentials#_custom_helpers) looks like this is the way to go:
![image](https://user-images.githubusercontent.com/26760571/107517449-6f45d700-6bb6-11eb-922f-7ecaef548601.png)